src/golang/lib/workflow/dag/workflow_dag.go

```diff
@@ -99,7 +98,7 @@ func NewWorkflowDag(
 			artifactIDToContentPath[artifactID],
 			artifactIDToMetadataPath[artifactID],
 			artifactResultWriter,
-			storageConfig,
+			&dbWorkflowDag.StorageConfig,
```

Comment:
I got a little confused here, since `dbWorkflowDag` makes me think that the storage config is something set by the user. However, it looks like this is the same storage config that is in `AqContext` right? 

If so, then this seems like something that will become clearer once we stop serializing requests into `dbWorkflowDag`s.

Reply:
And one more clarification: the storage config is set here at the beginning, and then written to the workflow spec. Subsequent scheduled runs will then use whatever is on the workflow spec, and that is immutable over the lifecycle  of the workflow?

Reply:
So the storage config is `dbWorkflowDag` is set based on the value of the global storage config at the time of the workflow registration. Once this value is set, we persist it to the database, and that storage config is used for the entire lifecycle of the workflow. Right now we don't support modifying the storage config for existing workflows.